src/sensor/ping.cpp

```diff
@@ -167,6 +167,7 @@ void Ping::addDetectionLink(LinkType connType, const QStringList& connString)
 void Ping::connectLink(LinkType connType, const QStringList& connString)
 {
     Sensor::connectLink(LinkConfiguration{connType, connString});
+    SettingsManager::self()->lastLinkConfiguration(*link()->configuration());
```

Comment:
I believe we should connect connectLink to connectionDetected, and move the first requests here.

Reply:
This is already done. This function is called by the QML part, because sensor does not provide it as Q_INVOKABLE. This is a result of a bug where QML can't handle functions with default values or with the same name.